src/dotty/tools/dotc/transform/Space.scala

```diff
@@ -0,0 +1,316 @@
+package dotty.tools.dotc
+package transform
+
+import core.Types._
+import core.Contexts._
+import core.Flags
+import ast.Trees._
+import ast.tpd
+import core.Decorators._
+
+
+/** space logic for checking exhaustivity and unreachability of
+  * pattern matching.
+  */
+
+
+// space definition
+sealed trait Space
```

Comment:
This is pretty cool: you're using the exhaustiveness algorithm to test its own exhaustivity! :collision: